coderd/coderd.go

```diff
@@ -592,7 +592,14 @@ func New(options *Options) *API {
 		r.Route("/users", func(r chi.Router) {
 			r.Get("/first", api.firstUser)
 			r.Post("/first", api.postFirstUser)
-			r.Get("/authmethods", api.userAuthMethods)
+			r.Route("/authmethods", func(r chi.Router) {
+				// The API Key allows this method to return the auth method
+				// for the logged-in user. This information is useful for the
+				// caller. If not authenticated, this information is omitted.
+				r.Use(apiKeyMiddlewareOptional)
+				r.Get("/", api.userAuthMethods)
+			})
```

Comment:
I agree with the idea of only returning for /users/me rather than hacking it on top of the auth methods API

Reply:
The downside to `/users/me` is that I need to make a new `codersdk.User` type with an optional field of the logintype. I do not want to leak this login type to non-admins. 

So it just becomes a refactor of that type. I can do it, I was just avoiding it.

Reply:
The `authmethods` felt ok since it's used to render the available login method types. If you are authenticated, this would tell you which one you are using. 

I believe it does make sense, and isn't a "hack". But understand wanting it on the `User` type. We currently don't have 2 types of users for the extra "private" fields.

Reply:
Make bespoke /user/me/logintype